src/library/scala/collection/concurrent/TrieMap.scala

```diff
@@ -988,6 +992,8 @@ object TrieMap extends MapFactory[TrieMap] {
 
   val inodeupdater = AtomicReferenceFieldUpdater.newUpdater(classOf[INodeBase[_, _]], classOf[MainNode[_, _]], "mainnode")
 
+  private[concurrent] val NoSuchElementSentinel: AnyRef = new AnyRef
```

Comment:
The `RESTART` sentinel is defined in a separate Java class. Don't know if it really makes a difference, but just saying...